dev/tools/gen_localizations.dart

```diff
@@ -107,19 +107,74 @@ String generateString(String s) {
   return output.toString();
 }
 
+/// Simple data class to hold parsed locale. Does not promise validity of any data.
+class Locale {
```

Comment:
It would be best to call this class something else, maybe `LocaleInfo`. Just to avoid confusion with Flutter's Locale.